docs/source/concepts/calibration/PowderDiffractionCalibration.rst

```diff
@@ -317,6 +317,181 @@ saved with :ref:`SaveDiffCal <algm-SaveDiffCal>`.
+
+    jsonfile = "/SNS/users/y8z/Temp/autogrouping_config.json"
+    with open(jsonfile, 'r') as jf:
+        config = json.load(jf)
+    # execute
+    main(config)
+
+An example json file is presented below to control the grouping behavior,
+
+.. code-block:: json
+
+    {
+        "DiamondFile": "/SNS/NOM/IPTS-24637/nexus/NOM_144974.nxs.h5",
+        "MaskFile": "/SNS/users/y8z/Temp/mask144974.out",
+
+        "GroupingMethod": "KMEANS_ED",
+        "NumberOutputGroups": "4",
+        "StandardScaling": false,
+
+        "FittingFunctionParameters": "Mixing,Intensity,PeakCentre,FWHM",
+        "FitPeaksArgs": { "PeakFunction": "PseudoVoigt",
+                        "PeakParameterNames": "Mixing",
+                        "PeakParameterValues": "0.6",
+                        "HighBackground": false,
+                        "MinimumPeakHeight": 3,
+                        "ConstrainPeakPositions": false
+                        },
+        "DiamondPeaks": "0.8920,1.0758,1.2615",
+        "ParameterThresholds": { "PeakCentre": "(0.01,10.0)",
+                                "Height": "(0.0,10000.0)"
+                            },
+
+        "FilterByChi2": { "Enable": true,
+                        "Value": 1e4
+                        },
+
+
+        "OutputGroupingFile": "./outputgrouping.xml",
+        "OutputMaskFile": "./outputmask.txt",
+
+        "OutputFitParamFile": "./outputfitparamtable.nxs",
+
+        "CacheDir": "./tmp/",
+
+        "Plots": { "Grouping": true,
+                "ED_Features": true,
+                "PCA": true,
+                "KMeans_Elbow": true,
+                "KMeans_Silhouette": true}
+    }
+
+
+and description for entries in the input json file is summarized in the following table,
+
+.. list-table::
+    :widths: 25 50
+    :header-rows: 1
+
+    * - Name
+      - Description
+    * - DiamondFile
+      - Full name of the input nexus file. For calibration purpose, usually a diamond measurement will be used.
+    * - MaskFile
+      - Full name of the input mask file. The file should contain a whole bunch of lines with a single integter in each line specifying the detector ID to be masked (index starting from 0).
+    * - GroupingMethod
+      - The method to be used for grouping. Valid input could be 'KMEANS_CC', 'KMEANS_DG', 'KMEANS_ED', 'DBSCAN_CC', 'DBSCAN_DG' and 'DBSCAN_ED'. 'KMEANS' and 'DBSCAN' refers to the two clustering methods. The second part of those values refers to the method for calculating similarity between spectra. 'CC' for cross-correlation, 'DG' for De Gelder similarity and 'ED' for Euclidean distance in parameter space.
+    * - NumberOutputGroups
+      - The number of groups to cluster all input spectra into. If using 'DBSCAN' method, there is no need to specify this parameter.
+    * - StandardScaling
+      - Whether or not to scale the input spectra by removing the mean and scaling to unit variance before clustering.
+    * - WorkspaceIndexRange
```

Comment:
@Kvieta1990 I don't think this option is being used in the current autogrouping script, it may be worth double-checking this?

Reply:
I added in this parameter in the autogrouping script, but I forgot to submit PR to `mantidtotalscattering`. will do that. Thanks a lot for bringing this up!